TribalWars/src/WebApi/Controllers/RegisterController.cs

```diff
@@ -0,0 +1,58 @@
+using System.Threading.Tasks;
+using ApplicationCore.Contract.Requests;
+using ApplicationCore.Contract.Responses;
+using ApplicationCore.DTOs;
+using ApplicationCore.Interfaces;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/auth")]
+    public class RegisterController : ControllerBase
+    {
+        private readonly IUserService _userService;
+        private readonly IMapper _mapper;
+
+        public RegisterController(IUserService userService, IMapper mapper)
+        {
+            _userService = userService;
+            _mapper = mapper;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> CreateUser([FromBody] CreateUserRequest request)
+        {
+            if (await _userService.UserExistsByEmailAsync(request.Email))
+                return BadRequest(new ErrorsResponse("user with this email already exists"));
```

Comment:
You can use IStringLocalizer to store and retrieve user messages. I think Bartek has already configured it.